### ISSUE TITLE
Adding components option (buttons)

### ISSUE THREAD

**Issue by Serpensin** [fieu/discord.sh#45]
Is it possible for you to add the option to send buttons (that lead to a website)?

**Comment by fieu** [comment]
There might be a possibility to add buttons. Not sure if you can send them within webhooks.

**Comment by fieu** [comment]
Hello @Serpensin,

I took a look at the possibility of adding message components support. As far as I can tell, they can only be used in “application-owned” webhooks (interactions).

Here’s my request just in case someone wants to take a crack at it, however I think they are not supported in basic webhook integrations for the moment. They just lead to empty messages when sent.

![image](https://user-images.githubusercontent.com/6765260/216924188-64e8e776-0818-4b4e-9019-e18f6c390d1e.png)

```sh
curl -X "POST" "https://discord.com/api/webhooks/XX/XX?wait=true" \
     -H 'Content-Type: application/json; charset=utf-8' \
     -d $'{
  "components": [
    {
      "type": 1,
      "components": [
        {
          "label": "Example",
          "style": 5,
          "type": 2,
          "url": "https://example.com"
        }
      ]
    }
  ]
}'
```

Closing this issue for now. If they get supported, feel free to open a new issue.
